test/test_torch.py

```diff
@@ -2811,6 +2811,34 @@ def test_randperm(self):
         self.assertEqual(res1.numel(), 0)
         self.assertEqual(res2.numel(), 0)
 
+    @unittest.skipIf(not torch.cuda.is_available(), 'no CUDA')
+    def test_randperm_cuda(self):
+        cuda = torch.device('cuda:0')
+
+        # For small inputs, randperm is offloaded to CPU instead
+        _RNGState_cpu = torch.get_rng_state()
+        _RNGState_cuda = torch.cuda.get_rng_state()
```

Comment:
You could just use `torch.random.fork_rng` (with `devices=[0]` to make it faster)